### ISSUE TITLE
Product attribute values changed after updating product in the app

### ISSUE THREAD

**Issue by reginabally** [woocommerce/woocommerce-android#3847]
**Describe the bug**

It was reported in 3909132-zd-woothemes that editing the stock quantity of a product in the app will cause the attribute values to change (separated by commas).

**To Reproduce**
1. Go to Products tab in the app, select a product that has custom attributes (contain more than one word) added
2. Edit the stock quantity of the product
3. Notice commas (,) will be added after each word in the attributes
4. Checking the attribute of the product on WP Admin, notice each word in the attributes are separated by | (originally reported in 3903610-zd-woothemes)

**Screen recording**

https://user-images.githubusercontent.com/40906847/114806547-892eb080-9dd7-11eb-96d1-21303e3236f6.mp4

**Expected behavior**

I would expect the product attributes remain unchanged.

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: OPPO Reno 4
- Android version: 11
- WooCommerce Android version: 6.4-rc-2

**WordPress Environment**
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://woo20200908.mystagingwebsite.com
Site address (URL): https://woo20200908.mystagingwebsite.com
WC Version: 5.2.0
REST API Version: ✔ 5.2.0
WC Blocks Version: ✔ 4.7.2
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 2.1.5
Log Directory Writable: ✔
WP Version: 5.7
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.3.27
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.75.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.24-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.2.0
WC Database Prefix: wp_
Total Database Size: 7.19MB
Database Data Size: 5.00MB
Database Index Size: 2.19MB
wp_woocommerce_sessions: Data: 0.23MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mo2f_network_blocked_ips: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_email_sent_audit: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_transactions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_whitelisted_ips: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_user_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_user_login_info: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 0.30MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.22MB + Index: 0.14MB + Engine InnoDB
wp_posts: Data: 0.14MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfblockediplog: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfblocks7: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfconfig: Data: 0.47MB + Index: 0.00MB + Engine InnoDB
wp_wfcrawlers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilechanges: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilemods: Data: 0.42MB + Index: 0.00MB + Engine InnoDB
wp_wfhits: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wp_wfhoover: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfknownfilelist: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_wflivetraffichuman: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wflocs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wflogins: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfnotifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfpendingissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfreversecache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfsnipcache: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfstatus: Data: 0.09MB + Index: 0.03MB + Engine InnoDB
wp_wftrafficrates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_attack_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_backup_report: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_files_scan: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpns_ip_rate_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_hash_file: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wpns_malware_scan_report: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_scan_report_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_skip_files: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yoast_indexable: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.02MB + Index: 0.03MB + Engine InnoDB

### Post Type Counts ###

attachment: 59
page: 10
post: 6
product: 31
product_variation: 23
revision: 10
seedprod: 1
shop_coupon: 1
shop_order: 25
shop_order_refund: 3
wc_order_status: 7

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (2) ###

Jetpack by WordPress.com: by Automattic – 9.6.1
WooCommerce: by Automattic – 5.2.1

### Inactive Plugins (16) ###

Akismet Anti-Spam: by Automattic – 4.1.9
All in One Product Quantity for WooCommerce: by WPWhale – 3.7.2
Classic Editor: by WordPress Contributors – 1.6
Coming Soon Page, Maintenance Mode & Landing Pages by SeedProd: by SeedProd – 6.2.1
Custom Order Status for WooCommerce: by Tyche Softwares – 2.0.2
Enable jQuery Migrate Helper: by The WordPress Team – 1.3.0
Extra Product Options (Product Addons) for WooCommerce: by ThemeHigh – 3.0.2
miniOrange 2 Factor Authentication: by miniOrange – 5.4.33
Under Construction: by WebFactory Ltd – 3.88
WooCommerce Order Status Manager: by SkyVerge – 1.13.0
WooCommerce Product Add-ons: by WooCommerce – 3.3.1
WooCommerce Shipment Tracking: by WooCommerce – 1.6.26
WooCommerce Stripe Gateway: by WooCommerce – 5.1.0
Wordfence Security: by Wordfence – 7.5.2
XML-RPC Validator: by Danilo E – 0.1
Yoast SEO: by Team Yoast – 16.1.1

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #104 - /
Cart: #105 - /cart/
Checkout: #106 - /checkout/
My account: #107 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.5.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 24
Oldest: 2021-04-06 00:38:24 +0000
Newest: 2021-04-15 02:36:01 +0000


### Status report information ###

Generated at: 2021-04-15 02:46:42 +00:00
`
```
</details>

**Comment by anitaa1990** [comment]
PR with a fix for this in https://github.com/woocommerce/woocommerce-android/pull/3840

**Comment by anitaa1990** [comment]
Fix released in v6.5-rc-1 🎉
